providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
Well, my regexp simply removes a ending '(/.*)?' or '.*'. Remove all the escapes (backslashes) and you will see.
How about that: We add restorecon as an optional parameter and extend the current regexp by other patterns from "semanage fcontext -l". If we make restorecon a separate provider we could even produce nice log output so people can see, and debug, what goes wrong.
Is have some spare time tomorrow and could implement that. Getting the no_regex_filename would then of course be a library definition.